app/experimenter/experiments/models/nimbus.py

```diff
@@ -276,7 +276,23 @@ def targeting(self):
 
         if self.locales.count():
             locales = [locale.code for locale in self.locales.all().order_by("code")]
-            expressions.append(f"locale in {locales}")
+
+            if self.application == self.Application.DESKTOP:
+                expressions.append(f"locale in {locales}")
+            else:
+                iso_locale_list = [locale for locale in locales if "-" in locale]
+                iso_locales = set([locale[:2] for locale in iso_locale_list])
+                iso_locales_expression = " || ".join(
+                    [f"'{language}' in locale" for language in list(iso_locales)]
+                )
+                iso_languages = [locale for locale in locales if "-" not in locale]
+                iso_languages_expression = " || ".join(
+                    [f"'{language}' in locale" for language in iso_languages]
+                )
+                locale_expressions = (
+                    f"({iso_locales_expression} || {iso_languages_expression})"
+                )
```

Comment:
I think you'll find if you take out the `"-" in locale` check on line 283 you can take this second case out altogether.